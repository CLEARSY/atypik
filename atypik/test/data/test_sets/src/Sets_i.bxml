<?xml version="1.0" encoding="UTF-8" ?>
<Machine xmlns='https://www.atelierb.eu/Formats/bxml' version='1.0' name='Sets_i' type='implementation' position='true' semantic='true' b0check='false'>
<Abstraction>Sets</Abstraction>
<Values>
<Attr>
<Pos l='9' c='1' s='6'/>
</Attr>
<Valuation ident='ABS_ENS' typref='2'>
<Attr>
<Pos l='10' c='5' s='7'/>
</Attr>
<Binary_Exp op='..' typref='2'>
<Attr>
<Pos l='10' c='16' s='2'/>
</Attr>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='10' c='15' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='100' typref='1'>
<Attr>
<Pos l='10' c='18' s='3'/>
</Attr>
</Integer_Literal>
</Binary_Exp>
</Valuation>
</Values>
<Concrete_Variables>
<Attr>
<Pos l='12' c='1' s='18'/>
</Attr>
<Id value='var' typref='1'>
<Attr>
<Pos l='13' c='5' s='3'/>
</Attr>
</Id>
</Concrete_Variables>
<Invariant>
<Attr>
<Pos l='15' c='1' s='9'/>
</Attr>
<Exp_Comparison op=':'>
<Attr>
<Pos l='16' c='9' s='1'/>
</Attr>
<Id value='var' typref='1'>
<Attr>
<Pos l='16' c='5' s='3'/>
</Attr>
</Id>
<Id value='ABS_ENS' typref='2'>
<Attr>
<Pos l='16' c='11' s='7'/>
</Attr>
</Id>
</Exp_Comparison>
</Invariant>
<Initialisation>
<Attr>
<Pos l='18' c='1' s='14'/>
</Attr>
<Assignement_Sub>
<Attr>
<Pos l='19' c='9' s='2'/>
</Attr>
<Variables>
<Id value='var' typref='1'>
<Attr>
<Pos l='19' c='5' s='3'/>
</Attr>
</Id>
</Variables>
<Values>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='19' c='12' s='1'/>
</Attr>
</Integer_Literal>
</Values>
</Assignement_Sub>
</Initialisation>
<TypeInfos>
<Type id='0'>
<Id value='BOOL'/>
</Type>
<Type id='1'>
<Id value='INTEGER'/>
</Type>
<Type id='2'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Type>
<Type id='3'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Unary_Exp>
</Type>
</TypeInfos>

</Machine>