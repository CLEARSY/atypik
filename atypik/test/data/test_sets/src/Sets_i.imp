/* Sets_i
 * Author: vdemeulenaere
 * Creation date: 14/06/2023
 */

IMPLEMENTATION Sets_i
REFINES Sets

VALUES 
    ABS_ENS = 1..100

CONCRETE_VARIABLES 
    var

INVARIANT 
    var : ABS_ENS

INITIALISATION 
    var := 1

 END