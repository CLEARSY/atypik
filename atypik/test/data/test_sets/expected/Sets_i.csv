Expression	Identifier	Line	Column	Type
"ABS_ENS = (1) .. (100)"	75	10	5	(POW INTEGER)
"100"	73	10	18	INTEGER
"(1) .. (100)"	74	10	16	(POW INTEGER)
"ABS_ENS"	4	16	11	(POW INTEGER)
"var"	76	13	5	INTEGER
"1"	72	10	15	INTEGER
"1"	77	19	12	INTEGER
